builtin/providers/cloudstack/tags.go

```diff
@@ -0,0 +1,81 @@
+package cloudstack
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+// tagsSchema returns the schema to use for tags.
+//
+func tagsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Optional: true,
+	}
+}
+
+// setTags is a helper to set the tags for a resource. It expects the
+// tags field to be named "tags"
+func setTags(cs *cloudstack.CloudStackClient, d *schema.ResourceData, resourcetype string) error {
+
+	if d.HasChange("tags") {
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		create, remove := diffTags(tagsFromMap(o), tagsFromMap(n))
```

Comment:
This would be useful when either adding one extra tag, removing only one (out of more) or modifying one. The worst case of these would be when modifying one where it would remove it and create it anew (api limitation), but on all other cases, we would not be removing and creating the untouched tags. 

In example 2 in the PR description, it would do:
```
createTag("test_tag2=false")
``` 
instead of:
```
removeTag("test_tag=true")
createTag("test_tag=true")
createTag("test_tag2=false")
```
as the sorting would remove the test_tag=true from the create group and would not put it in the remove group, but would put test_tag2=false on the create group